python/lsst/daf/persistence/butler.py

```diff
@@ -606,6 +606,14 @@ def get(self, datasetType, dataId=None, immediate=False, **rest):
             bypassFunc = getattr(location.mapper, "bypass_" + datasetType)
             callback = lambda: bypassFunc(datasetType, pythonType, location, dataId)
         else:
+            if isinstance(location, ButlerComposite):
+                for name, info in location.components.items():
+                    location.components[name] = self.get(info.datasetType, location.dataId, immediate=True, 
+                                                         **rest)
```

Comment:
`**rest` has already been added to the dataId, so it shouldn't need to be passed through here.